tests/test_search.py

```diff
@@ -145,26 +145,24 @@ def test_custom(self):
         index.add_vectors(np.eye(5, dtype=np.float32))
         self.assertIsInstance(index.faiss_index, faiss.IndexFlat)
 
-    def test_serialization(self):
-        import faiss
-
-        index = FaissIndex(metric_type=faiss.METRIC_INNER_PRODUCT)
-        index.add_vectors(np.eye(5, dtype=np.float32))
-
-        # Setting delete=False and unlinking manually is not pretty... but it is required on Windows to
-        # ensure somewhat stable behaviour. If we don't, we get PermissionErrors. This is an age-old issue.
-        # see https://bugs.python.org/issue14243 and
-        # https://stackoverflow.com/questions/23212435/permission-denied-to-write-to-my-temporary-file/23212515
-        with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
-            index.save(tmp_file.name)
-            index = FaissIndex.load(tmp_file.name)
-        os.unlink(tmp_file.name)
 
-        query = np.zeros(5, dtype=np.float32)
-        query[1] = 1
-        scores, indices = index.search(query)
-        self.assertGreater(scores[0], 0)
-        self.assertEqual(indices[0], 1)
+@require_faiss
+def test_serialization(mockfs):
```

Comment:
IMO the function above should be kept to still run the test to make sure that local storage works, while you can rename this to something like:

```suggestion
def test_serialization_fs(mockfs):
```

To be compliant with other functions like e.g. https://github.com/huggingface/datasets/blob/cc637d107ef3e3b9948691379312a8099b6476aa/tests/test_arrow_dataset.py#L3655